akka-projection-grpc/src/main/scala/akka/projection/grpc/internal/ProtoAnySerialization.scala

```diff
@@ -167,16 +175,62 @@ import scalapb.options.Scalapb
         else
           idAndManifest.substring(0, i).toInt -> idAndManifest.substring(i + 1)
 
+      println(s"# deserialize 3: $typeUrl, akka $id $manifest") // FIXME
+
       serialization.deserialize(scalaPbAny.value.toByteArray, id, manifest).get
     } else if (prefer == Prefer.Scala) {
+      println(s"# deserialize 4: $typeUrl") // FIXME
       // when custom typeUrl
       scalaPbAny
     } else {
+      println(s"# deserialize 5: $typeUrl") // FIXME
       // when custom typeUrl
       ScalaPbAny.toJavaProto(scalaPbAny)
     }
   }
 
+  def toSerializedEvent(scalaPbAny: ScalaPbAny): Option[SerializedEvent] = {
+    // see corresponding typeUrl cases in `deserialize`
+
+    def pbAnyToSerializedEvent(): SerializedEvent = {
+      val pbAny =
+        if (prefer == Prefer.Scala) scalaPbAny
+        else ScalaPbAny.toJavaProto(scalaPbAny)
+      val serializer = serialization.findSerializerFor(pbAny)
+      val manifest = Serializers.manifestFor(serializer, pbAny)
+      SerializedEvent(scalaPbAny.toByteArray, serializer.identifier, manifest)
+    }
+
+    val typeUrl = scalaPbAny.typeUrl
+    if (typeUrl == ProtoAnyTypeUrl) {
+      Some(pbAnyToSerializedEvent())
+    } else if (typeUrl.startsWith(GoogleTypeUrlPrefix)) {
+      val messageClass = resolveTypeUrl(typeUrl).messageClass
+      serialization.serializerFor(messageClass) match {
+        case _: SerializerWithStringManifest =>
+          // we can't find the manifest without creating an instance, i.e. Serializers.manifestFor can't be used here
+          None
+        case serializer =>
+          // this is the case for akka.remote.serialization.ProtobufSerializer, which is configured by default
+          // for protobuf messages
+          val manifest = if (serializer.includeManifest) messageClass.getName else ""
+          Some(SerializedEvent(scalaPbAny.toByteArray, serializer.identifier, manifest))
```

Comment:
or is it `scalaPbAny.value.toByteArray` - I will write some tests to get this right
and maybe this is only safe for the Akka ProtobufSerializer